src/hermes/commands/deposit/invenio_rdm.py

```diff
@@ -0,0 +1,717 @@
+
+    licenses_api_path = invenio_config.get("api_paths", {}).get(
+        "licenses", _DEFAULT_LICENSES_API_PATH
+    )
+    licenses_api_url = f"{site_url}/{licenses_api_path}"
+    license = _get_license_identifier(ctx, licenses_api_url)
+    ctx.update(invenio_path["license"], license)
+
+    communities_api_path = invenio_config.get("api_paths", {}).get(
+        "communities", _DEFAULT_COMMUNITIES_API_PATH
+    )
+    communities_api_url = f"{site_url}/{communities_api_path}"
+    communities = _get_community_identifiers(ctx, communities_api_url)
+    ctx.update(invenio_path["communities"], communities)
+
+    access_right, embargo_date, access_conditions = _get_access_modalities(license)
+    ctx.update(invenio_path["access_right"], access_right)
+    ctx.update(invenio_path["embargo_date"], embargo_date)
+    ctx.update(invenio_path["access_conditions"], access_conditions)
+
+
+def map_metadata(click_ctx: click.Context, ctx: CodeMetaContext):
+    """Map the harvested metadata onto the Invenio schema."""
+
+    deposition_metadata = _codemeta_to_invenio_deposition(ctx)
+
+    metadata_path = ContextPath.parse("deposit.invenio_rdm.depositionMetadata")
+    ctx.update(metadata_path, deposition_metadata)
+
+    # Store a snapshot of the mapped data within the cache, useful for analysis, debugging, etc
+    with open(ctx.get_cache("deposit", "invenio_rdm", create=True), 'w') as invenio_json:
+        json.dump(deposition_metadata, invenio_json, indent='  ')
+
+
+def create_initial_version(click_ctx: click.Context, ctx: CodeMetaContext):
+    """Create an initial version of a publication.
+
+    If a previous publication exists, this function does nothing, leaving the work for
+    :func:`create_new_version`.
+    """
+
+    invenio_path = ContextPath.parse("deposit.invenio_rdm")
+    invenio_ctx = ctx[invenio_path]
+    latest_record_id = invenio_ctx.get("latestRecord", {}).get("id")
+
+    if latest_record_id is not None:
+        # A previous version exists. This means that we need to create a new version in
+        # the next step. Thus, there is nothing to do here.
+        return
+
+    if not click_ctx.params['initial']:
+        raise RuntimeError("Please use `--initial` to make an initial deposition.")
+
+    _log = logging.getLogger("cli.deposit.invenio_rdm")
+
+    invenio_config = config.get("deposit").get("invenio_rdm", {})
+    site_url = invenio_config["site_url"]
+    depositions_api_path = invenio_config.get("api_paths", {}).get(
+        "depositions", _DEFAULT_DEPOSITIONS_API_PATH
+    )
+
+    deposition_metadata = invenio_ctx["depositionMetadata"]
+
+    deposit_url = f"{site_url}/{depositions_api_path}"
+    response = requests.post(
+        deposit_url,
+        json={"metadata": deposition_metadata},
+        headers={
+            "User-Agent": hermes_user_agent,
+            "Authorization": f"Bearer {click_ctx.params['auth_token']}",
+        }
+    )
+
+    if not response.ok:
+        print(response.text)
```

Comment:
Add this to the `RuntimeError` or use logging instead?

Reply:
**joke:** And I though we only want to avoid *print**f*** debugging...